tests/runtime/conftest.py

```diff
@@ -149,6 +161,47 @@ def runtime_grpc_server(
         yield server
 
 
+@contextmanager
+def runtime_http_test_server(open_port, *args, **kwargs):
+    """Helper to wrap creation of RuntimeHTTPServer in temporary configurations"""
+    with tempfile.TemporaryDirectory() as workdir:
+        temp_log_dir = os.path.join(workdir, "metering_logs")
+        temp_save_dir = os.path.join(workdir, "training_output")
+        os.makedirs(temp_log_dir)
+        os.makedirs(temp_save_dir)
+        with temp_config(
+            {
+                "runtime": {
+                    "metering": {"log_dir": temp_log_dir},
+                    "training": {"output_dir": temp_save_dir},
+                    "http": {"port": open_port},
+                }
+            },
+            "merge",
+        ):
+            with http_server.RuntimeHTTPServer(*args, **kwargs) as server:
+                # Give tests access to the workdir
+                server.workdir = workdir
+                yield server
+
+
+# This is honestly not required for fastAPI testing
```

Comment:
If we don't actually need this, let's not keep it in. That said, we _do_ want to be able to test the real server behavior to ensure that our server booting and binding works (particularly with (m)TLS). I know we have tests for that somewhere, so either they should use this, or we should remove this if there's a compelling reason to not use this (which I think there is since they want the server to boot/terminate in isolation).

Reply:
Ok, I see that you've refactored those tests to do `runtime_http_test_server`, so let's remove this one

Reply:
I'm thinking I can actually add a test case that uses this to make actual REST calls to the server?

Reply:
This also removes the duplicated code snippet from all tests
```
server = http_server.RuntimeHTTPServer()
```